services/scheduler/src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java

```diff
@@ -85,6 +88,7 @@
   private org.quartz.Scheduler quartzScheduler;
 
   private volatile boolean started = false;
+  private List<Scheduler> activeSchedulers = synchronizedList(new ArrayList<>());
```

Comment:
who clears this list? it looks like you're speculating with schedulers only be disposed when the application is disposed and that's not always true. Sources are a good counter example. Custom extensions is another. These schedulers should be decorated so that they're removed from this list when shutdown